Combine SexualOrientationPreference with ContentPreferenceValue

A-haron opened the pull request:
- What is the purpose of the pull request?

Eliminate Redundancy between SexualOrientationPreference and ContentPreferenceValue.

- Give a brief description of what you changed or added.

Removed SexualOrientationPreference, moved the only unique method to SexualOrientation (And created a generic one for ContentPreferenceValue), and replaced all references.

- Are any new graphical assets required?

None.

- Has this change been tested? If so, mention the version number that the test was based on.

Yes, with latest /dev. (v0.3.2.6 preview release)

- So we have a better idea of who you are, what is your Discord Handle?

@A-haron#3467

- If you want quick feedback, you can use @Innoxia, or jump on the Lilith's Throne discord and send Innoxia a PM.

Look at it once you have time, no need to hurry ;-)

A-haron commented:
Just noted that I didn't put a space between @ and Innoxia. Sorry for the tag ("_";)

Rfpnj commented:
Let me see if I understand what you have here. You changed the method of setting Sexual Orientation to be a percentage that can be adjusted individually. So a person could see say 90% M/M content and 10% M/F if they set the numbers accordingly.

A-haron commented:
ContentPreferenceValue and SexualOrientationPreference are already basically the same (You can check both files [here](https://github.com/Innoxia/liliths-throne-public/blob/af43087e88a27e0a793a61f548d2e5c8bf373d70/src/com/lilithsthrone/game/character/persona/SexualOrientationPreference.java) and [here](https://github.com/Innoxia/liliths-throne-public/blob/af43087e88a27e0a793a61f548d2e5c8bf373d70/src/com/lilithsthrone/game/settings/ContentPreferenceValue.java)), except for the `SexualOrientationPreference::getSexualOrientationFromUserPreferences` method.

I generalized the method (So you can now generate a random weighted value from any ContentPreferenceValue), moved the specific version of it to SexualOrientation, and changed references.
